src/misc/debug.h

```diff
@@ -39,13 +43,80 @@ extern void debug_vprintf_trace(const char* func_name,
 				unsigned int line,
 				int level, const char* fmt, va_list ap);
 
+#ifdef __cplusplus
+#  define EXTERN_FUNC extern "C"
+#else
+#  define EXTERN_FUNC extern
+#endif /* __cplusplus */
+
+#ifdef BART_SUPER_DEBUG
```

Comment:
we already have debug_level to control the amount of debug output.

Reply:
I will remove this and replace it with calls to `debug_printf`